pybamm/experiment/experiment.py

```diff
@@ -42,13 +42,17 @@ class Experiment:
 
     def __init__(
         self,
-        operating_conditions,
-        period="1 minute",
-        temperature=None,
-        termination=None,
+        operating_conditions: list,
+        period: str = "1 minute",
+        temperature: float = None,
+        termination: list = None,
         drive_cycles=None,
         cccv_handling=None,
     ):
+        if not (isinstance(operating_conditions, list)):
+            raise TypeError("operating_conditions must be list of strings. For example:"
+                            f"\n\n [{operating_conditions}]")
+
```

Comment:
Could you convert to a string to a list of strings here, based on https://github.com/pybamm-team/PyBaMM/issues/3623#issue-2043497552 again – and raise an error on any other object? Also, you would need to improve the test coverage by adding a test for the condition you added

Reply:
Type checks should handle it already. It should just throw an error instead of trying to recover from bad inputs

Reply:
Type checks are basically just hints in Python. Not everyone has them enabled in their IDEs.

You could check for a list of strings using something like -
```py
all(isinstance(cond, operating_conditions) for cond in operating_conditions)
```

Reply:
We could return the helpful error, but we could equally just have it well documented. We probably don't need to handle every possible type of error.

Reply:
Agreed